src/main/java/io/cdap/e2e/pages/locators/CdfSysAdminLocators.java

```diff
@@ -51,4 +53,108 @@ public class CdfSysAdminLocators {
+
+  @FindBy(how = How.XPATH, using = "//a[@href='/cdap/httpexecutor']")
+  public static WebElement makeHttpCall;
+
+  //clicking on create new namespace button
+  @FindBy(how = How.XPATH, using = "//button[@data-testid='create-namespace-btn']")
+  public static WebElement createNewNamespace;
+
+  //clicking on edit system preference button
+  @FindBy(how = How.XPATH, using = "//button[@data-testid='edit-system-prefs-btn']")
+  public static WebElement editSystemPreference;
+
+  //clicking on create new computeProfile
+  @FindBy(how = How.XPATH, using = "//a[@href='/cdap/ns/system/profiles/create']")
+  public static WebElement createComputeProfile;
+
+  //clicking on namespace or system preference heading
+  public static WebElement clickPreference(String preferences) {
+    String path = "//div[@data-cy='" + preferences + "']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(path));
+  }
+
+  @FindBy(how = How.XPATH, using = "//div//label[@for='import-profile']")
+  public static WebElement clickImport;
+
+  @FindBy(how = How.XPATH, using = "//button[@data-testid='save-prefs-btn']")
+  public static WebElement clickSaveClose;
+
+  @FindBy(how = How.XPATH, using = "//div[@data-testid='create-namespace-name']")
+  public static WebElement namespaceName;
+
+  @FindBy(how = How.XPATH, using = "//input[@placeholder='Namespace description' and @type='text']")
+  public static WebElement namespaceDescription;
+
+  @FindBy(how = How.XPATH, using = "//button[@data-cy='wizard-next-btn']")
+  public static WebElement clickNext;
+
+  public static WebElement locateButtonAddNamespace(String buttonType) {
+    String xpath = "//*[@data-cy='wizard-" + buttonType + "-btn']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  @FindBy(how = How.XPATH, using = "//div[contains(@class,'namespaces')]")
+  public static WebElement openNamespace;
+
+  @FindBy(how = How.XPATH, using = "//div[contains(@class,'system-profiles')]")
+  public static WebElement openSystemComputeProfile;
+
+  public static WebElement locateKeyProperty(String pluginProperty, int row) {
+    String xpath = "//*[@data-cy='" + pluginProperty + "" + row + "']//input[@placeholder='key']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateValueProperty(String pluginProperty, int row) {
+    String xpath = "//*[@data-cy='" + pluginProperty + "" + row + "']//input[@placeholder='value']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateAddRowButtonProperty(String pluginProperty, int row) {
+    String xpath = "//*[@data-cy='" + pluginProperty + "" + row + "']//span//button[@type='submit' and " +
+      "@class='btn add-row-btn btn-link']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  @FindBy(how = How.XPATH, using = "//span[contains(@class,'reset')]")
+  public static WebElement clickReset;
```

Comment:
If we are click on a button to reset it. For better naming convention and readability, can we rename it to **clickOnResetButton**.

Reply:
Bharat , for sysAdmin changes the PR is already reviewed and now it is in External review , not changing anything related to sysAdmin . I have rebased my branch to sysAdmin as there are some common locators / action class dependencies .